app/user/rest.py

```diff
@@ -197,6 +197,8 @@ def verify_user_code(user_id):
 
     user_to_verify.current_session_id = str(uuid.uuid4())
     user_to_verify.logged_in_at = datetime.utcnow()
+    if user_to_verify.auth_type == 'email_auth':
```

Comment:
By the way, these two lines of code here may be similar to the way I was suggesting in my above larger comment. Each individual view is responsible for deciding whether the action has been done by email and therefore bumping the last validated at field.